ambari-server/src/main/java/org/apache/ambari/server/serveraction/upgrades/DeleteUnsupportedServicesAndComponents.java

```diff
@@ -68,7 +72,10 @@ public CommandReport execute(ConcurrentMap<String, Object> requestSharedDataCont
   private Set<String> deleteUnsupportedServices(Cluster cluster, RepositoryVersionEntity repoVersion) throws AmbariException {
     Set<String> servicesToBeRemoved = serviceComponentSupport.unsupportedServices(cluster, repoVersion.getStackName(), repoVersion.getStackVersion());
     for (String serviceName : servicesToBeRemoved) {
-      cluster.deleteService(serviceName, new DeleteHostComponentStatusMetaData());
+      DeleteHostComponentStatusMetaData deleteMetaData = new DeleteHostComponentStatusMetaData();
+      cluster.deleteService(serviceName, deleteMetaData);
+      STOMPComponentsDeleteHandler.processDeleteByMetaDataException(deleteMetaData);
+      STOMPComponentsDeleteHandler.processDeleteByMetaData(deleteMetaData);
```

Comment:
These changes aren't very intuitive - I'm afraid that someone else using the cluster.deleteService(...) call would miss these. Why can't we make them a part of that call?

Reply:
@jonathan-hurley this was done to reduce amount of topology update events.
If we do this implicitly, any high-level delete (service, cluster) will produce swarm of update events, up to few hundreds which affects performance a lot.
I agree that there should be better way, but imo it will require some refactoring and should be done as separate task.

Reply:
we can add extra argument like - sendUpdateEvent?